ledger/participant-local-store/src/main/scala/com/daml/platform/localstore/CachedIdentityProviderConfigStore.scala

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.localstore
+
+import com.daml.caching.CaffeineCache
+import com.daml.caching.CaffeineCache.FutureAsyncCacheLoader
+import com.daml.ledger.api.domain.{IdentityProviderConfig, IdentityProviderId}
+import com.daml.logging.LoggingContext
+import com.daml.metrics.Metrics
+import com.daml.platform.localstore.api.IdentityProviderConfigStore.Result
+import com.daml.platform.localstore.api.{IdentityProviderConfigStore, IdentityProviderConfigUpdate}
+import com.github.benmanes.caffeine.{cache => caffeine}
+
+import java.time.Duration
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Success, Try}
+
+class CachedIdentityProviderConfigStore(
+    delegate: IdentityProviderConfigStore,
+    expiryAfterWriteInSeconds: Int,
+    maximumCacheSize: Int,
+    metrics: Metrics,
+)(implicit val executionContext: ExecutionContext, loggingContext: LoggingContext)
+    extends IdentityProviderConfigStore {
+
+  private val idpCache: CaffeineCache.AsyncLoadingCaffeineCache[
+    IdentityProviderId.Id,
+    Result[IdentityProviderConfig],
+  ] =
+    new CaffeineCache.AsyncLoadingCaffeineCache(
+      caffeine.Caffeine
+        .newBuilder()
+        .expireAfterWrite(Duration.ofSeconds(expiryAfterWriteInSeconds.toLong))
+        .maximumSize(maximumCacheSize.toLong)
+        .buildAsync(
+          new FutureAsyncCacheLoader[IdentityProviderId.Id, Result[IdentityProviderConfig]](
+            delegate.getIdentityProviderConfig
+          )
+        ),
+      metrics.daml.identityProviderConfigStore.cache,
+    )
+
+  private val idpListCache: CaffeineCache.AsyncLoadingCaffeineCache[
+    CaffeineCache.type,
```

Comment:
Is this an established pattern? Maybe instead create a custom object like `CachedIdentityProviderListSingletonKey` (or something less verbose)